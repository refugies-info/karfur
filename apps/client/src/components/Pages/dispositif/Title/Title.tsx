import { useContext } from "react";
import { useSelector } from "react-redux";
import { selectedDispositifSelector } from "~/services/SelectedDispositif/selectedDispositif.selector";
import PageContext from "~/utils/pageContext";
import { TitleEdit } from "../Edition";
import Text from "../Text";
import styles from "./Title.module.scss";

interface Props {}

/**
 * Shows a text as title, or the form component of the Title. Can be used for VIEW or EDIT mode.
 */
const Title = () => {
  const dispositif = useSelector(selectedDispositifSelector);
  const pageContext = useContext(PageContext);

  return (
    <h1 className={styles.title}>
      {pageContext.mode !== "edit" ? (
        <Text id="titreInformatif">{dispositif?.titreInformatif || ""}</Text>
      ) : (
        <TitleEdit />
      )}
    </h1>
  );
};

export default Title;
