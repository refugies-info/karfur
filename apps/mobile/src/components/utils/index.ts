export * from "./array";
export { isDarkColor } from "./isDarkColor";
