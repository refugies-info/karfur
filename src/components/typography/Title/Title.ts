import styled from "styled-components/native";
import { TextBigBold } from "../../StyledText";

const Title = styled(TextBigBold)`
  margin-bottom: ${({ theme }) => theme.margin * 2}px;
`;

export default Title;
