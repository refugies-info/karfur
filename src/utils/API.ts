import axios from "react-native-axios";
import { getEnvironment } from "../libs/getEnvironment";

const dbURL = getEnvironment().dbUrl;
const siteSecret = getEnvironment().siteSecret;

const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});

export const getLanguages = () => apiCaller.get("/langues/getLanguages");

export const getCitiesFromGoogleAPI = (text: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${text}&language=fr&types=(cities)&components=country:fr&key=${process.env.GOOGLE_API_KEY}`
  );

export const getCityDetailsFromGoogleAPI = (placeId: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/details/json?place_id=${placeId}&fields=address_component&key=${process.env.GOOGLE_API_KEY}`
  );

export const getPlaceIdFromLocationFromGoogleAPI = (
  longitude: number,
  latitude: number
) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${process.env.GOOGLE_API_KEY}`
  );

export const getContentsForApp = ({
  locale,
  age,
  department,
  frenchLevel,
}: {
  locale: string;
  age: string | null;
  department: string | null;
  frenchLevel: string | null;
}) => {
  const route =
    `/dispositifs/getContentsForApp?locale=${locale}` +
    (age ? `&age=${age}` : "") +
    (department ? `&department=${department}` : "") +
    (frenchLevel ? `&frenchLevel=${frenchLevel}` : "");

  return apiCaller.get(route);
};

export const get_dispositif = (params = {}) =>
  apiCaller.post("/dispositifs/get_dispositif", params);
