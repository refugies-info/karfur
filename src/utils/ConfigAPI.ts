import axios from "react-native-axios";
import Config from "../libs/getEnvironment";

const dbURL = Config.dbUrl;
const siteSecret = Config.siteSecret;
export const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});