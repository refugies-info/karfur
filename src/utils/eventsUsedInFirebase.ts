export const events = {
  REACHED_END_CONTENT: "REACHED_END_CONTENT",
  LANGUAGE_VALIDATE: "LANGUAGE_VALIDATE",
  LOCATION_VALIDATE: "LOCATION_VALIDATE",
};
