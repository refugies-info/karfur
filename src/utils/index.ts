export { default as withProps } from "./withProps";
