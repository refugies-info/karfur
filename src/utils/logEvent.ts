/* eslint-disable no-console */
import * as Analytics from "expo-firebase-analytics";
import { getEnvironment } from "../libs/getEnvironment";
import { FirebaseEvent } from "./eventsUsedInFirebase";

export const logEventInFirebase = async (
  eventName: FirebaseEvent,
  data: Record<string, any>
) => {
  const { envName, debugModeFirebase } = getEnvironment();
  if (envName === "DEVELOPMENT") {
    if (debugModeFirebase === "activated") {
      Analytics.setDebugModeEnabled(true);
      console.log(
        "Environment is dev and debug mode is activated for firebase"
      );
    } else {
      console.log(
        "Environment is dev and debug mode is NOT activated for firebase"
      );
      return;
    }
  }

  if (envName && ["DEVELOPMENT", "STAGING", "PROD"].includes(envName)) {
    await Analytics.logEvent(eventName, { ...data, envName });
  }
};
