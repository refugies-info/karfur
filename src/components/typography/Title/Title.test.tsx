import React from "react";
import Title from "./Title";
import { render } from "../../utils/tests";

describe("Title snapshot test suite", () => {
  it("should render without bug", async () => {
    const test = await render(<Title>Mon titre</Title>);
    expect(test).toMatchSnapshot();
  });
});
