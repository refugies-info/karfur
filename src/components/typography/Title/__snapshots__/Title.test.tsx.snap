// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title snapshot test suite should render without bug 1`] = `
<Text
  isRTL={false}
  selectable={true}
  style={
    Array [
      Object {
        "color": "#212121",
        "fontFamily": "circularBold",
        "fontSize": 25,
        "lineHeight": 32,
        "marginBottom": 16,
        "textAlign": "left",
      },
    ]
  }
>
  Mon titre
</Text>
`;
